mysql/tests/test_mysql.py

```diff
@@ -803,6 +803,7 @@ def test_database_instance_metadata(aggregator, dd_run_check, instance_complex,
     expected_tags = tags.METRIC_TAGS + [
         "database_hostname:{}".format(expected_database_hostname),
         "database_instance:{}".format(expected_database_instance),
+        "dbms_flavor:{}".format(MYSQL_FLAVOR.lower()),
```

Comment:
This seems like an actual bug in our current logic. We set `dbms_flavor` tag on all of our metrics/events except for `database_instance` event. So then we also miss it on any cloud provider dynamic metric linking as well. This would fix that